process/interceptors/processor/trieNodeChunksProcessor.go

```diff
@@ -1,3 +1,233 @@
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/batch"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/interceptors/processor/chunk"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+const minimumRequestTimeInterval = time.Second
+
+type chunkHandler interface {
+	Put(chunkIndex uint32, buff []byte)
+	TryAssembleAllChunks() []byte
+	GetAllMissingChunkIndexes() []uint32
+	Size() int
+	IsInterfaceNil() bool
+}
+
+type checkRequest struct {
+	batch        *batch.Batch
+	chanResponse chan process.CheckedChunkResult
+}
+
+// TrieNodesChunksProcessorArgs is the argument DTO used in the trieNodeChunksProcessor constructor
+type TrieNodesChunksProcessorArgs struct {
+	Hasher          hashing.Hasher
+	ChunksCacher    storage.Cacher
+	RequestInterval time.Duration
+	RequestHandler  process.RequestHandler
+	Topic           string
+	ShardID         uint32
+}
+
+type trieNodeChunksProcessor struct {
+	hasher            hashing.Hasher
+	chunksCacher      storage.Cacher
+	chanCheckRequests chan checkRequest
+	requestInterval   time.Duration
+	requestHandler    process.RequestHandler
+	topic             string
+	shardID           uint32
+	cancel            func()
+	chanClose         chan struct{}
+}
+
+// NewTrieNodeChunksProcessor creates a new trieNodeChunksProcessor instance
+func NewTrieNodeChunksProcessor(arg TrieNodesChunksProcessorArgs) (*trieNodeChunksProcessor, error) {
+	if check.IfNil(arg.Hasher) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilHasher)
+	}
+	if check.IfNil(arg.ChunksCacher) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilCacher)
+	}
+	if arg.RequestInterval < minimumRequestTimeInterval {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor, minimum request interval is %v",
+			process.ErrInvalidValue, minimumRequestTimeInterval)
+	}
+	if check.IfNil(arg.RequestHandler) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilRequestHandler)
+	}
+	if len(arg.Topic) == 0 {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrEmptyTopic)
+	}
+
+	tncp := &trieNodeChunksProcessor{
+		hasher:            arg.Hasher,
+		chunksCacher:      arg.ChunksCacher,
+		chanCheckRequests: make(chan checkRequest),
+		requestInterval:   arg.RequestInterval,
+		requestHandler:    arg.RequestHandler,
+		topic:             arg.Topic,
+		shardID:           arg.ShardID,
+		chanClose:         make(chan struct{}),
+	}
+	var ctx context.Context
+	ctx, tncp.cancel = context.WithCancel(context.Background())
+	go tncp.processLoop(ctx)
+
+	return tncp, nil
+}
+
+func (proc *trieNodeChunksProcessor) processLoop(ctx context.Context) {
+	chanDoRequests := time.After(proc.requestInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("trieNodeChunksProcessor.processLoop go routine is stopping...")
```

Comment:
do you really want this debug ? :) if you want to test it real time, you should delete the protection code from smartContract process.go

Reply:
We have debug prints like this on all go routines as to easily check during debugging sessions